testenv/aws/aws.go

```diff
@@ -0,0 +1,243 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aws
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	cloudformationtypes "github.com/aws/aws-sdk-go-v2/service/cloudformation/types"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+
+	"github.com/openclarity/vmclarity/testenv/aws/asset"
+	"github.com/openclarity/vmclarity/testenv/types"
+	"github.com/openclarity/vmclarity/testenv/utils"
+	"github.com/openclarity/vmclarity/testenv/utils/docker"
+)
+
+type ContextKeyType string
+
+const (
+	AWSClientContextKey         ContextKeyType = "AWSClient"
+	DefaultStackCreationTimeout                = 30 * time.Minute
+	DefaultSSHPortReadyTimeout                 = 2 * time.Minute
+	DefaultRemoteUser                          = "ubuntu"
+)
+
+var _ types.Environment = &AWSEnv{}
+
+// AWS Environment.
+type AWSEnv struct {
+	client              *cloudformation.Client
+	ec2Client           *ec2.Client
+	s3Client            *s3.Client
+	testAsset           *asset.Asset
+	server              *Server
+	stackName           string
+	templateURL         string
+	workDir             string
+	region              string
+	sshKeyPair          *utils.SSHKeyPair
+	sshPortForwardInput *utils.SSHForwardInput
+	meta                map[string]interface{}
+
+	*docker.DockerHelper
+}
+
+type Server struct {
+	InstanceID string
+	PublicIP   string
+}
+
+func (s Server) WaitForSSH(ctx context.Context, t time.Duration) error {
+	// TODO(paralta): Implement method to check if SSH port is ready
+	time.Sleep(t) // nolint:gomnd
+
+	return nil
+}
+
+// Setup AWS test environment from cloud formation template.
+// * Create a new CloudFormation stack from template
+// (upload template file to S3 is required since the template is larger than 51,200 bytes).
+// * Create test asset.
+// * When infrastructure is ready, start SSH port forwarding and remote docker client.
+func (e *AWSEnv) SetUp(ctx context.Context) error {
+	// Prepare stack
+	err := e.prepareStack(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to prepare stack: %w", err)
+	}
+
+	// Create a new CloudFormation stack from template
+	_, err = e.client.CreateStack(
+		ctx,
+		&cloudformation.CreateStackInput{
+			StackName:    &e.stackName,
+			Capabilities: []cloudformationtypes.Capability{cloudformationtypes.CapabilityCapabilityIam},
+			TemplateURL:  &e.templateURL,
+			Parameters: []cloudformationtypes.Parameter{
+				{ParameterKey: aws.String("KeyName"), ParameterValue: &e.stackName},
```

Comment:
multiline split for clarity
```suggestion
				{
					ParameterKey: aws.String("KeyName"), 
					ParameterValue: &e.stackName,
				},
```